### PR TITLE
[#2601] Add Getting Started guide with Kafka based messaging.

### PR DESCRIPTION
**Pull request by b-abel**
This adds an adapted Getting Started guide for Kafka-based messaging. The diagram showing the containers involved in an example deployment is also adapted.
~It references the documentation of deploying Hono with Kafka in the _IoT Packages_ project, which is included in a [PR](eclipse/packages#237) in the repository there. This PR should be merged after eclipse/packages#237.~

### CODE REVIEW

**Comment by b-abel** [comment]
This PR is currently blocked by #2683 and eclipse/packages#237.

**Comment by b-abel** [comment]
@sophokles73 I'm interested in your opinion on the question raised in https://github.com/eclipse/packages/pull/237#issuecomment-849560483. Which of the following options do you prefer? 

1. Investing time to configure Kafka so that it allows northbound clients to connect with TLS or plaintext depending on the port (as it is currently the case with the qdrouter) or 
2. support only TLS connections for Kafka and document (in this guide here) how to configure northbound clients to connect with TLS. It should be quite easy to provide a second `values.yaml` file with the Helm chart that can be used as a "profile" to deploy a plaintext setup instead.

I would prefer option 2. I don't see very much value in enabling TLS and plaintext at the same time. The Kafka deployment on the Sandbox on the other hand could be plaintext only because FMPOV it is mainly used for the Getting Started guide, which currently only uses plaintext connections.

**Comment by sophokles73** [comment]
@b-abel I agree, we should start with option 2.

**Comment by b-abel** [comment]
This is now ready to be reviewed.

**Comment by b-abel** [comment]
Note that merging this PR will instantly (i.e. with the next run of the homepage deployment) make the content available on Hono's landing page.

**Comment by b-abel** [comment]
I assume that Kafka-based messaging will be "fully supported" from version 1.10.0 of Hono. Therefore I suggest merging this PR with the release of that version.

File: site/homepage/content/getting-started-kafka/index.md [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+kubectl get secrets eclipse-hono-kafka-jks --template="{{index .data \"kafka.truststore.jks\" | base64decode}}" -n hono > $KAFKA_TRUSTSTORE_PATH
+~~~
+
+and then start the client (make sure to replace `my-tenant`):
+
+~~~sh
+# in directory where the hono-cli-*-exec.jar file has been downloaded to
+export MY_TENANT=my-tenant
+export KAFKA_IP=$(kubectl get service eclipse-hono-kafka-0-external --output="jsonpath={.status.loadBalancer.ingress[0]['hostname','ip']}" -n hono)
+java -jar hono-cli-*-exec.jar --hono.kafka.commonClientConfig.bootstrap.servers=$KAFKA_IP:9092 --hono.kafka.commonClientConfig.security.protocol=SASL_SSL --hono.kafka.commonClientConfig.sasl.jaas.config="org.apache.kafka.common.security.scram.ScramLoginModule required username=\"hono\" password=\"hono-secret\";" --hono.kafka.commonClientConfig.sasl.mechanism=SCRAM-SHA-512 --spring.profiles.active=receiver,kafka --tenant.id=$MY_TENANT --hono.kafka.commonClientConfig.ssl.truststore.location=$KAFKA_TRUSTSTORE_PATH --hono.kafka.commonClientConfig.ssl.truststore.password=honotrust --hono.kafka.commonClientConfig.ssl.endpoint.identification.algorithm=""
```
Comment (sophokles73) [inline site/homepage/content/getting-started-kafka/index.md:254]: this looks very complicated :-( Are you sure that it still reflects the command client client's current state/requirements?

Comment (b-abel) [reply site/homepage/content/getting-started-kafka/index.md:254]: Alternatively, we could skip all the parameters for the authentication and use the sandbox profile from the CLI that already contains them as defaults. Would you prefer that?

Comment (sophokles73) [reply site/homepage/content/getting-started-kafka/index.md:254]: Wasn't that the overall idea when introducing the profile(s)?

Comment (b-abel) [reply site/homepage/content/getting-started-kafka/index.md]: Please have a look at my latest commit.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**